src/odemis/dataio/tiff.py

```diff
@@ -1206,28 +1206,55 @@ def _findImageGroups(das):
     current_ifd = 0
     prev_da = None
 
+    # required keys for merging
+    required_keys = [model.MD_IN_WL, model.MD_OUT_WL]
+    matching_keys = {
+        model.MD_HW_NAME: None,
+        model.MD_HW_VERSION: None,
+        model.MD_PIXEL_SIZE: None,
+        model.MD_POS: None,
+        model.MD_ROTATION: 0,
+        model.MD_SHEAR: 0,
+    }
     for da in das:
         # check if it can be part of the current group (compare just to the previous DA)
         dims = da.metadata.get(model.MD_DIMS, "CTZYX"[-da.ndim::])
-        if (prev_da is None
-            or da.shape[0] != 1  # If C != 1 => not possible to merge (C is always first dimension) or it is a
-                # greyscale YX image
-            or (model.MD_IN_WL not in da.metadata or model.MD_OUT_WL not in da.metadata)
+
+
+        plen_c, len_c = 1, 1
+        if "C" in dims:
+            len_c = da.shape[dims.index("C")]
+        if prev_da is not None:
+            prev_dims = prev_da.metadata.get(model.MD_DIMS, "CTZYX"[-prev_da.ndim::])
+            if "C" in prev_dims:
+                plen_c = prev_da.shape[prev_dims.index("C")]
```

Comment:
Instead of explicitly computing it, you can do the same as `prev_da`: start with `plen_c = None`, and at the end of the loop body, do a `plen_c = len_c`.